### ISSUE TITLE
Server defined post length not respected

### ISSUE THREAD

**Issue by RobATX** [mastodon/mastodon-ios#1049]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Universeodon currently has message limit set to 1,000 characters. The iOS app shows 500, which is the typical default. App should use whatever length is set for the server. 

### Expected Behavior

Use server defined post length/limit

### Environment

```markdown
- Device:
- OS:
- Version:
- Build:
```

**Comment by j-f1** [comment]
Where in the UI are you seeing this? In the in-app post composer UI or in the share sheet → Mastodon option? We’re supposed to be pulling the limit from the API but clearly that isn’t being wired up correctly. Also, does it go away once you start typing?

**Comment by RobATX** [comment]
In the composer. It shows 500 at the bottom. ![image](https://github.com/mastodon/mastodon-ios/assets/31712535/2dbc907f-afac-46b0-841d-fb3bfcf5f80d)

**Comment by vmstan** [comment]
Seeing this now on vmst.io after we raised our limit to 512. Still shows 500 in the app.

**Comment by vmstan** [comment]
This appears to be related to a change introduced in v1.4.7

https://github.com/mastodon/mastodon-ios/blob/1.4.7/MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/ComposeContentViewModel.swift

```
   // limit
    @Published public var maxTextInputLimit = 500
```

**Comment by vmstan** [comment]
I have confirmed by adjusting this value you can control the post length but I'm not sure how to import the value that server presents in Swift.

**Comment by 100000BC** [comment]
I would love to see this get worked on. I am not a coder so I have no idea how to fix it. Does making a new comment here boost the issue? :)

**Comment by zeitschlag** [comment]
Closed in #1207
